src/System.Management.Automation/DscSupport/CimDSCParser.cs

```diff
@@ -2534,7 +2534,13 @@ private static void ProcessMembers(StringBuilder sb, List<object> embeddedInstan
 
                 string arrayAffix = isArrayType ? "[]" : string.Empty;
 
-                sb.Append(CultureInfo.InvariantCulture, $"    {MapAttributesToMof(enumNames, attributes, embeddedInstanceType)}{mofType} {member.Name}{arrayAffix};\n");
+                sb.AppendFormat(
+                    CultureInfo.InvariantCulture,
+                    "    {0}{1} {2}{3};\n",
+                    MapAttributesToMof(enumNames, attributes, embeddedInstanceType),
+                    mofType,
+                    member.Name,
+                    arrayAffix);
```

Comment:
This will only increase the number of allocations in the parser and make it slower.

Reply:
In this particular case, `sb.AppendFormat` will incur an array allocation because there are 4 objects passed in.
Updated the code to continue using `sb.Append(`, but in a readable manner.

Reply:
Such allocations will are removed in .Net 8.0 (for all APIs with `params`) so you can revert to `sb.AppendFormat` if you prefer.

Reply:
But a question, @iSazonov, if .NET 8 is going to support 4 or more formating objects in `string.Format` or `StringBuilder.AppendFormat`, then will most of the changes to `string.Create(<interpolated-string>)` be in vain?

Reply:
This will only remove allocating array for arguments. One more optimization.
Interpolated strings already use InterpolatedStringHandler which utilizes stackalloc to avoid allocations at all in formatting process. Also it excludes parsing format string again and again. (They added new API CompositeFormat on this week to cache format parsing.)